File: src/main/java/akostenko/aicars/keyboard/SingleKeyAction.java
package akostenko.aicars.keyboard;

import org.newdawn.slick.Input;
import org.newdawn.slick.KeyListener;

import java.util.function.Consumer;

public class SingleKeyAction implements KeyListener {
    private final int key;
    private final Consumer<Void> action;

    public SingleKeyAction(Consumer<Void> action, int key) {
        this.key = key;
        this.action = action;
    }

    @Override
    public void keyPressed(int key, char c) {
        if (this.key == key) {
            action.accept(null);
        }
    }

    @Override
    public void keyReleased(int key, char c) {

    }

    @Override
    public void setInput(Input input) {

    }

    @Override
    public boolean isAcceptingInput() {
        return true;
    }

    @Override
    public void inputEnded() {

    }

    @Override
    public void inputStarted() {

    }
}

File: game-engine/src/main/kotlin/akostenko/aicars/keyboard/SingleKeyAction.kt
package main.kotlin.akostenko.aicars.keyboard

import org.newdawn.slick.Input
import org.newdawn.slick.KeyListener

import java.util.function.Consumer

class SingleKeyAction(private val action: () -> Unit, private val key: Int) : KeyListener {

    override fun keyPressed(key: Int, c: Char) {
        if (this.key == key) {
            action()
        }
    }

    override fun keyReleased(key: Int, c: Char) {

    }

    override fun setInput(input: Input) {

    }

    override fun isAcceptingInput(): Boolean {
        return true
    }

    override fun inputEnded() {

    }

    override fun inputStarted() {

    }
}
